lib/cdo/shared_constants/pd/teacher1819_application_constants.rb

```diff
@@ -206,11 +206,11 @@ def self.clean_multiline(string)
 
   CRITERIA_SCORE_QUESTIONS_CSP = (
     VALID_SCORES.select {|_, v| v == YES_NO}.keys - [:csd_which_grades, :csd_terms_per_year]
-  ).freeze
+  )
   CRITERIA_SCORE_QUESTIONS_CSD = (
     VALID_SCORES.select {|_, v| v == YES_NO}.keys -
       [:csp_how_offer, :csp_which_grades, :csp_course_hours_per_year]
-  ).freeze
+  )
```

Comment:
? Why unfreeze? These are constants, so they should be immutable. I noticed you are (correctly) [cloning](https://github.com/code-dot-org/code-dot-org/pull/20484/files#diff-71cf096063a13bbf44392ad72d8e471bR509) them so you can mutate the local copies. The underlying constants should remain frozen.

Reply:
OIC, Try `dup` instead of `clone` to get an unfrozen copy.

Reply:
Alternatively, don't dup/clone and don't mutate the array but reassign the reference to a modified copy when needed:

```ruby
irb(main):013:0> a = [1,2].freeze
=> [1, 2]
irb(main):014:0> a = a - [2]
=> [1]
irb(main):015:0> a
=> [1]
```

Reply:
This is super helpful! thank you! I had a feeling I would get advice on a better way to do this in review :)